Mage.Sets/src/mage/cards/c/CaptainAmericaFirstAvenger.java

```diff
@@ -0,0 +1,215 @@
+
+    // Functional negation of AnotherPredicate.
+    @Override
+    public boolean apply(ObjectSourcePlayer<MageObject> input, Game game) {
+        if (!input.getObject().getId().equals(input.getSourceId())) {
+            return false;
+        }
+        int zcc = input.getSource().getSourceObjectZoneChangeCounter();
+        return zcc == input.getObject().getZoneChangeCounter(game);
+    }
+
+    @Override
+    public String toString() {
+        return "{this}";
+    }
+}
+
+enum CaptainAmericaFirstAvengerValue implements DynamicValue {
+    instance;
+
+    @Override
+    public int calculate(Game game, Ability sourceAbility, Effect effect) {
+        int amount = 0;
+        for (Cost cost : sourceAbility.getCosts()) {
+            if (cost instanceof CaptainAmericaFirstAvengerUnattachCost && !cost.getTargets().isEmpty()) {
+                Permanent equipment = game.getPermanentOrLKIBattlefield(cost.getTargets().getFirstTarget());
+                if (equipment != null) {
+                    amount = equipment.getManaValue();
+                }
+            }
+        }
+        return amount;
+    }
+
+    @Override
+    public DynamicValue copy() {
+        return instance;
+    }
+
+    @Override
+    public String toString() {
+        return "X";
+    }
+
+    @Override
+    public String getMessage() {
+        return "that Equipment's mana value";
+    }
+}
+
+class CaptainAmericaFirstAvengerUnattachCost extends EarlyTargetCost {
+
+    private static final FilterPermanent filter = new FilterEquipmentPermanent("equipment attached to this creature");
+    private static final FilterPermanent subfilter = new FilterControlledPermanent("{this}");
+
+    static {
+        subfilter.add(CaptainAmericaPredicate.instance);
+        filter.add(new AttachedToPredicate(subfilter));
+    }
+
+    public CaptainAmericaFirstAvengerUnattachCost() {
+        super();
+    }
+
+    protected CaptainAmericaFirstAvengerUnattachCost(final CaptainAmericaFirstAvengerUnattachCost cost) {
+        super(cost);
+    }
+
+    @Override
+    public boolean canPay(Ability ability, Ability source, UUID controllerId, Game game) {
+        Permanent permanent = game.getPermanent(source.getSourceId());
+        return permanent != null
+                && !permanent.getAttachments().isEmpty();
+    }
+
+    @Override
+    public boolean pay(Ability ability, Game game, Ability source, UUID controllerId, boolean noMana, Cost costToPay) {
+        Player player = game.getPlayer(source.getControllerId());
+        Permanent permanent = game.getPermanent(source.getSourceId());
+        if (permanent == null) {
+            return paid;
+        }
+        Permanent equipment = game.getPermanentOrLKIBattlefield(getTargets().getFirstTarget());
+        if (equipment == null || !permanent.getAttachments().contains(getTargets().getFirstTarget()) ||
+                !player.chooseUse(Outcome.Benefit, "Unattach " + equipment.getName() + "?", source, game)) {
```

Comment:
`getIdName` (in case two equipment with the same name are attached)

Reply:
Done.